ci/windows-pkg-arrow-for-r.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
```

Comment:
I think it would be critical to add `set -eu`, otherwise this script can silently generates error with our CI capturing it. For example, boost download could fail and the script would move on and repacking the zip without boost.

Reply:
I can add that but it's not critical; if this script doesn't do everything it's supposed to, the next step in the Appveyor job will fail. (Trust me, I learned that the hard way ;)

Reply:
If the script executes correctly with `set -eu` right now, future you will thank past you :)